app/renderer/components/navigation/navigationBar.js

```diff
@@ -199,80 +283,136 @@ class NavigationBar extends React.Component {
         : null
       }
       {
-        this.props.titleMode
-        ? null
-        : this.props.loading
-          ? <span className='navigationButtonContainer'>
-            <button data-l10n-id='stopButton'
-              className='normalizeButton navigationButton stopButton'
-              onClick={this.onStop} />
-          </span>
-          : <span className='navigationButtonContainer'>
-            <LongPressButton
-              l10nId='reloadButton'
-              className='normalizeButton navigationButton reloadButton'
-              onClick={this.onReload}
-              onLongPress={this.onReloadLongPress} />
-          </span>
+        !this.props.titleMode
+        ? <span className={css(
+            commonStyles.navigationButtonContainer,
+            styles.navigationBar__navigationButtonContainer,
+          )}>
+          {
+            this.props.loading
+            ? this.stopButton
+            : this.reloadButton
+          }
+        </span>
+        : null
       }
       {
         !this.props.titleMode && getSetting(settings.SHOW_HOME_BUTTON)
-        ? <span className='navigationButtonContainer'>
-          <button
-            data-test-id='homeButton'
-            data-l10n-id='homeButton'
-            className='normalizeButton navigationButton homeButton'
-            onClick={this.onHome} />
+        ? <span className={css(
+            commonStyles.navigationButtonContainer,
+            styles.navigationBar__navigationButtonContainer,
+          )}>
+          {this.homeButton}
         </span>
         : null
       }
-      <div className='startButtons'>
-        {
-          !this.props.titleMode
-          ? <span className='bookmarkButtonContainer'>
-            <button data-l10n-id={this.bookmarked ? 'removeBookmarkButton' : 'addBookmarkButton'}
-              className={cx({
-                navigationButton: true,
-                bookmarkButton: true,
-                removeBookmarkButton: this.bookmarked,
-                withHomeButton: getSetting(settings.SHOW_HOME_BUTTON),
-                normalizeButton: true
-              })}
-              onClick={this.onToggleBookmark} />
-          </span>
-          : null
-        }
-      </div>
+      {
+        !this.props.titleMode
+        ? this.bookmarkButtonContainer
+        : null
+      }
       <UrlBar
         titleMode={this.props.titleMode}
         onStop={this.onStop}
         menubarVisible={this.props.menubarVisible}
-        />
+      />
       {
-        isSourceAboutUrl(this.props.location)
-        ? null
-        : <div className='endButtons'>
-          {
-            <PublisherToggle
-              location={this.props.location}
-              locationInfo={this.props.locationInfo}
-              siteSettings={this.props.siteSettings}
-              synopsis={this.props.synopsis}
-            />
-          }
-        </div>
+        !this.props.titleMode && !isSourceAboutUrl(this.props.location)
```

Comment:
publisher toggle has its own rules which are kind of complicated. this check should work from there. I think only check for titleMode here is better. If we want to add a container just for this component it should live in the component itself